### PR TITLE
Move auth to use directives

### PR DESCRIPTION
**Pull request by jack-mccabe-defra**

### CODE REVIEW

File: test/auth/authenticate.test.js [hunk 1]
```diff
@@ -61,3 +68,55 @@ describe('getAuth', () => {
     expect(mockPublicKeyFunc).toHaveBeenCalledWith(undefined)
   })
 })
+
+describe('checkAuthGroup', () => {
+  const adminGroupId = process.env.ADMIN_AD_GROUP_ID
+
+  it('checkAuthGroup should not throw an error for admins with correct group', () => {
+    const testGroup = 'ADMIN'
```
Comment (rkdefra) [inline test/auth/authenticate.test.js:76]: This const is not used. Should this line just be removed?

File: test/auth/authenticate.test.js [hunk 2]
```diff
@@ -61,3 +68,55 @@ describe('getAuth', () => {
     expect(mockPublicKeyFunc).toHaveBeenCalledWith(undefined)
   })
 })
+
+describe('checkAuthGroup', () => {
+  const adminGroupId = process.env.ADMIN_AD_GROUP_ID
```
Comment (rkdefra) [inline test/auth/authenticate.test.js:73]: Why get this from an env var? Doesn't this just complicate the test setup slightly, or this there a reason for not just using a value here?

File: test/integration/auth/unauthenticated.test.js [hunk 3]
```diff
@@ -29,4 +29,54 @@ describe('Query.customer without authorization header', () => {
...
+              }
+            }
+          }
+        }
+      `,
+      schema,
+      contextValue: unAuthedContext
+    })
+    expect(result.data.business).toBeNull()
+    expect(result.errors).toHaveLength(1)
```
Comment (rkdefra) [inline test/integration/auth/unauthenticated.test.js:53]: Does the error need to be unpacked to ensure the call is failing for the right reason? Same for following test too...

**Review by rkdefra — CHANGES_REQUESTED** [review]
A minor improvement (which the linter should have picked up), and a couple of questions...

File: test/auth/authenticate.test.js (hunk 2 above)
Comment (jack-mccabe-defra) [reply test/auth/authenticate.test.js:73]: just also makes sure it is coming from env var

File: test/integration/auth/unauthenticated.test.js (hunk 3 above)
Comment (jack-mccabe-defra) [reply test/integration/auth/unauthenticated.test.js:59]: good point, added

File: test/auth/authenticate.test.js (hunk 1 above)
Comment (jack-mccabe-defra) [reply test/auth/authenticate.test.js]: removed

**Review by rkdefra — APPROVED** [review]
Looks good!

**Pull request merged**